### DISCUSSION TITLE
mergerfs and autofs

### DISCUSSION THREAD

**Discussion by nsc80** [trapexit/mergerfs#1003] [Q&A] [answered]
Hello, everyone.
I would like to thank in advance for any help.

I'm running ubuntu server 20.04.3 LTS and using autofs to automatically mount my disks on demand. I'm quite a linux beginner, but actively learning. I'm using autofs because all of my data drives are in a 5 bay USB enclosuse.
My auto.master file is populated with:

```
#/misc  /etc/auto.misc
#/net   -hosts
+dir:/etc/auto.master.d
#/nfs4  /usr/sbin/fedfs-map-nfs4 nobind
#+auto.master
/media   /etc/auto.media --timeout=1800
```

My auto.media:
`BACKUPS         -fstype=ntfs,rw,allow_other,permissions            :/dev/disk/by-uuid/861A5B8C1A5B77DF`
`NAS             -fstype=ntfs,rw,allow_other,permissions            :/dev/disk/by-uuid/AA8034E68034BAA1`
`NAS2            -fstype=ntfs,rw,allow_other,permissions            :/dev/disk/by-uuid/5CEA26CBEA26A16E`
`NAS3            -fstype=ntfs,rw,allow_other,permissions            :/dev/disk/by-uuid/2EC8B23DC8B202DF`
`seedbox         -fstype=ntfs,rw,allow_other,permissions            :/dev/disk/by-uuid/4EE0B3AFE0B39C21`

I've tried several ways to create a pool with NAS, NAS2 and NAS3, with someting like 
`storage         -fstype=fuse.mergerfs,rw,allow_other,permissions   /media/NAS:/media/NAS2:/media/NAS3`
but with no sucess. I can only find examples using fstab.

Can anyone help me?
Thank you

**Comment by trapexit** [comment, marked as answer]
I've never used autofs so I can only suggest mimicking what is found  on https://help.ubuntu.com/community/Autofs#FUSE_based_file_systems

If I can find time I'll look into more but may I ask... why on demand? Do you plan to use it so little that you don't even want to bother mounting it unless in use?

**Reply by nsc80** [reply]
That was the first place I went. Thank you anyway.
I was able to mount, removing the "permissions" argument and changing the mapping:
`storage         -fstype=fuse.mergerfs,rw,allow_other      :/media/NAS\:/media/NAS2\:/media/NAS3`
It now works, going to "test mode" to understand the mechanics of mergerfs.
I'll have to read some more to understand the impact of removing that argument.
Well I use autofs because my 5 bay enclosure doesn't spin down the disks when idle if using fstab. Probably there's a better way to achieve this with fstab, but I'm still learning.

Thank you for your help. Happy New Year!
